lib/src/errors.rs

```diff
@@ -1,19 +1,60 @@
+use std::error;
+use std::fmt;
+
 use telegram_bot_raw;
 
-error_chain! {
-    foreign_links {
-        Url(::hyper::http::uri::InvalidUri);
-        Hyper(::hyper::Error);
-        Http(::hyper::http::Error);
-        InvalidHeaderValue(::hyper::http::header::InvalidHeaderValue);
-        Io(::std::io::Error);
+#[derive(Debug)]
+pub struct Error(ErrorKind);
+
+#[derive(Debug)]
+pub(crate) enum ErrorKind {
+    Raw(telegram_bot_raw::Error),
+    Hyper(hyper::Error),
+    Http(hyper::http::Error),
+    Io(std::io::Error),
+    InvalidMultipartFilename,
+}
+
+impl From<telegram_bot_raw::Error> for ErrorKind {
```

Comment:
Wouldn't it be better to make this directly `impl From<> for Error`? So that it reduces the boilerplate for the `.map_err(ErrorKind::from)?`